recipes/openmpi/all/conanfile.py

```diff
@@ -1,97 +1,259 @@
+        # Not adding it as it fails to be detected by ./configure in some cases.
+        # https://github.com/open-mpi/ompi/blob/v4.1.6/opal/mca/dl/dl.h#L20-L25
+        tc.configure_args.append("--with-libltdl=no")
+        tc.generate()
+
+        deps = AutotoolsDeps(self)
+        deps.generate()
+
+        if self.options.with_verbs:
+            # Needed for ./configure to find libibnetdisc.so
+            VirtualRunEnv(self).generate(scope="build")
+
+        # TODO: might want to enable reproducible builds by setting
+        #  $SOURCE_DATE_EPOCH, $USER and $HOSTNAME
+
+    def _patch_sources(self):
+        # Not needed and fails with v5.0 due to additional Python dependencies
+        save(self, os.path.join(self.source_folder, "docs", "Makefile.in"), "all:\ninstall:\n")
+        # Workaround for <cstddef> trying to include VERSION from source dir due to a case-insensitive filesystem on macOS
+        # Based on https://github.com/macports/macports-ports/blob/22dded99ae76a287f04a9685bbc820ecaa397fea/science/openmpi/files/patch-configure.diff
+        replace_in_file(self, os.path.join(self.source_folder, "configure"),
+                        "-I$(top_srcdir) ", "-idirafter$(top_srcdir) ")
 
     def build(self):
-        with tools.chdir(self._source_subfolder):
-            autotools = self._configure_autotools()
-            autotools.make()
+        self._patch_sources()
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy(pattern="LICENSE", src=self._source_subfolder, dst="licenses")
-        with tools.chdir(self._source_subfolder):
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "etc"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        autotools = Autotools(self)
+        autotools.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "etc"))
+        rmdir(self, os.path.join(self.package_folder, "res", "man"))
+        rm(self, "*.la", self.package_folder, recursive=True)
+        fix_apple_shared_install_name(self)
 
     def package_info(self):
-        self.cpp_info.libs = ['mpi', 'open-rte', 'open-pal']
+        # Based on https://cmake.org/cmake/help/latest/module/FindMPI.html#variables-for-using-mpi
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "MPI")
+        self.cpp_info.set_property("pkg_config_name", "_ompi-do-not-use")
```

Comment:
```suggestion
        # TODO: Use None when available as Conan feature.
        self.cpp_info.set_property("pkg_config_name", "_ompi-do-not-use")
```

Reply:
Only to understand: Conan client has a feature request to be able to avoid generating the default .pc. It's not defined yet, neither has a PR, but the idea is getting mass as more and more recipes need it. So, marking it to remember in the future, once available, will make our life easier to filter it with a grep or similar.